icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -683,7 +683,11 @@ void serialize(MeasureUnitImpl& impl, UErrorCode& status) {
     if (U_FAILURE(status)) {
         return;
     }
-    U_ASSERT(impl.identifier.isEmpty());
+    if (!impl.identifier.isEmpty()) {
```

Comment:
This is fine, but what code triggered the assertion failure?

Reply:
Some code in "MixedUnits part 2" - want to see it? I'll tidy up that branch for an initial review. (Still have to deal with skeleton generation failure.)

Reply:
Ooh, ouch: looks like I might have triggered this assertion failure only via my debug output, so removing that, I don't need this? :-)

    const MeasureUnitImpl &outputUnit = MeasureUnitImpl::forMeasureUnit(unit, temp, status);
    const MeasureUnitImpl *inputUnit = &outputUnit;
    ...  // If inputUnit is mixed, replace by the first unit
    // printf("inputUnit: %s\n", std::move(inputUnit->copy(status)).build(status).getIdentifier());
    // printf("outputUnit: %s\n", std::move(outputUnit.copy(status)).build(status).getIdentifier());